src/doc/src/reference/unstable.md

```diff
@@ -1665,6 +1666,66 @@ version = "0.1.0"
 workspace = true
 ```
 
+## gc
+
+* Tracking issue: TODO
+
+The `-Zgc` flag enables garbage-collection within cargo's global cache within the cargo home directory.
+This includes downloaded dependencies such as compressed `.crate` files, extracted `src` directories, registry index caches, and git dependencies.
+When `-Zgc` is present, cargo will track the last time any file was used, and then uses those timestamps to manually or automatically delete files that have not been used for a while.
+
+```sh
+cargo build -Zgc
+```
+
+Automatic deletion happens on commands that are already doing a significant amount of work,
+such as all of the build commands (`cargo build`, `cargo test`, `cargo check`, etc.), and `cargo fetch`.
+Automatic deletion is only done once per day.
+Automatic deletion is disabled if cargo is offline such as with `--offline` or `--frozen` to avoid deleting files that may need to be used if you are offline for a long period of time.
+
+Manual deletion can be done with the `cargo clean` command.
+Deletion of cache contents can be performed by passing one of the cache options:
+
+- `--gc` --- Performs the same garbage collection that is performed by the once-a-day automatic deletion.
+- `--max-src-age=DURATION` --- Deletes source cache files that have not been used since the given age.
+- `--max-crate-age=DURATION` --- Deletes crate cache files that have not been used since the given age.
+- `--max-index-age=DURATION` --- Deletes registry indexes that have not been used since then given age (including their `.crate` and `src` files).
+- `--max-git-co-age=DURATION` --- Deletes git dependency checkouts that have not been used since then given age.
+- `--max-git-db-age=DURATION` --- Deletes git dependency clones that have not been used since then given age.
+- `--max-download-age=DURATION` --- Deletes any downloaded cache data that has not been used since then given age.
+- `--max-src-size=SIZE` --- Deletes the oldest source cache files until the cache is under the given size.
+- `--max-crate-size=SIZE` --- Deletes the oldest crate cache files until the cache is under the given size.
+- `--max-git-size=SIZE` --- Deletes the oldest git dependency caches until the cache is under the given size.
+- `--max-download-size=SIZE` --- Deletes the oldest downloaded cache data until the cache is under the given size.
+
```

Comment:
Kind of wondering what will happen, if I set a limit of 500mb but a rust projects needs a git db that exceeds this limit.
Will this break the build?

Reply:
Ah sorry, only now saw he reply (too much github things going on at a time :/ )

My assumption was that the the automatic gc would trigger every time, lets say after running "cargo build"

so cargo build would 
1) download the 1 Gb git repo
2) build the project
3) run auto gc
3.1) notice that 1 gb  exceeds the limit of 500 mb
3.2) delete the git repo from the cache

=> a subsequent "cargo build" would have to download the source again then?

Reply:
For "cargo script", we'll need to GC their target directory, especially if the script no longer exists.

We could instead track all workspaces.  One way we could use this is we could look at all `Cargo.lock` files and pin the entries in the cache so they can't be GCed.  This should solve that problem.  Of course, I have no idea how fast/slow this might be so no idea if we'll actually do it.

Reply:
My question is not about the artifacts that are produced by a build but about the dependencies that are downloaded before a build :sweat_smile:

Reply:
Thats what I'm referring to: we scan all `Cargo.lock`s for registry entries in use and don't clean them up.